From 3a28c2b8bb823bbfdd376f7c251e01a97a6b44f0 Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Wed, 20 Jun 2018 17:20:01 +0100
Subject: [fix] hackerrank, 'time delta', python solution

-feb idx off-by-one error in leap year application
---
 .../time.delta.py                             | 24 ++++++++++++++-----
 1 file changed, 18 insertions(+), 6 deletions(-)

diff --git a/hackerrank/python.-.date.and.time.-.medium.-.time.delta/time.delta.py b/hackerrank/python.-.date.and.time.-.medium.-.time.delta/time.delta.py
index 3a15fab..a3f3ac1 100644
--- a/hackerrank/python.-.date.and.time.-.medium.-.time.delta/time.delta.py
+++ b/hackerrank/python.-.date.and.time.-.medium.-.time.delta/time.delta.py
@@ -102,8 +102,8 @@ class zdate:
             self.d = month_days[self.mn]
             if self.d == 31 and self.m == 12:
                 self.y -= 1
-        elif self.d > month_days[self.mn] and (
-             not self.m == 2 and self.d == 29 and is_leap(self.y)):
+        elif self.d > month_days[self.mn] and not \
+             (self.m == 1 and self.d == 29 and is_leap(self.y)):
             self.m = (self.m + 1) % 12
             self.mn = list(month_days.keys())[self.m]
             self.d = 1
@@ -173,13 +173,25 @@ def time_delta(dt1, dt2):
     print("[zd2] d: %s, m: %d, dy: %d" % (zd2.to_string(), zd2.m, zd2.dy))
 
     print("[result] %d" % (s_delta))
-    if str(s_delta)[0] == "4":
-        print("[target] 413962293")
-    else:
-        print("[target] 1155677269")
+#   if str(s_delta)[0] == "4":
+#       print("[target] 413962293")
+#   elif str(s_delta)[0] == "9":
+#       print("[target] 9683593930")
+#   else:
+#       print("[target] 1155677269")
 
     return str(s_delta)
 
+"""
+3
+Fri 11 Feb 2078 00:05:21 +0400
+Mon 29 Dec 2064 03:33:48 -1100
+Thu 20 Oct 2140 04:28:27 +0330
+Thu 06 Mar 2104 23:00:38 -0430
+Fri 20 Apr 1973 17:33:30 -0200
+Sun 29 Feb 2280 13:45:40 +0300
+"""
+
 if __name__ == '__main__':
     fptr = open(os.environ['OUTPUT_PATH'], 'w')
 
-- 
2.23.0

